### ISSUE TITLE
Bugs in formatter

### ISSUE THREAD

**Issue by cmnrd** [lf-lang/lingua-franca#1832]
I applied the formatter to the examples repo today (https://github.com/lf-lang/examples-lingua-franca/pull/48). Interestingly, the CI formatting failed after I have correctly applied lff to all files. I investigated this found multiple issues with our formatter:

- In [HelloWorldPhysicalAfter.lf](https://github.com/lf-lang/examples-lingua-franca/blob/48f2e88887bc187600c06fd3a7d6cff9bd7f1d20/C/src/DistributedHelloWorld/HelloWorldPhysicalAfter.lf) the `;` after `at localhost` was removed by lff. This, however, appears to be illegal syntax.

- In [Rosace.lf](https://github.com/lf-lang/examples-lingua-franca/blob/48f2e88887bc187600c06fd3a7d6cff9bd7f1d20/C/src/rosace/Rosace.lf) lff made the following change on the first run: 
  ```
  - // Documentation
  - @icon("Variables.png")
  - reactor Variables {}
  + @icon("Variables.png")  // Documentation
  + reactor Variables {
  }
  ```
  I am not sure if we really should move the comment here. But the actual problem is that, on the next run, lff does this:
  ```
  - @Icon("Variables.png")  // Documentation
  + @icon("Variables.png")  // Documentation Documentation
  ```

- In [RosaceController.lf](https://github.com/lf-lang/examples-lingua-franca/blob/48f2e88887bc187600c06fd3a7d6cff9bd7f1d20/C/src/rosace/RosaceController.lf), does not like the following:
  ```
    az_f = new Filter(
        period = filter_period,
        init_x1 = 0.0,
        init_x2 = 0.0,
        a = (0.169118914523145, -0.518588903229759),
        b = (0.229019233988375, 0.421510777305010)
    )
  ```
  it converts this to:
  ```
  az_f = new Filter(
    period = filter_period,
    init_x1 = 0.0,
    init_x2 = 0.0,
    a = {},
    b = {}
  )
  ```
  Notice the empty assignment of `a` and `b`.
- Interestingly, when running lff again, it decides to flatten the above output to
  ```
   az_f = new Filter(period = filter_period, init_x1 = 0.0, init_x2 = 0.0, a = {}, b = {})
  ```
  Although the initial conversion from the previous point is wrong, a formatter should not change the output again when run repeatedly.
- In [Snake.lf](https://github.com/lf-lang/examples-lingua-franca/blob/48f2e88887bc187600c06fd3a7d6cff9bd7f1d20/Rust/src/Snake/Snake.lf]), lff converts the following:
  ```
  state grid: SnakeGrid ({= SnakeGrid::new(grid_side, &snake) =}); // note that this one borrows snake temporarily
  ```
  to this:
  ```
  state grid: SnakeGrid = {=  // note that this one borrows snake temporarily
    SnakeGrid::new(grid_side, &snake)
  =}
  ```
  I would argue that this conversion shouldn't be legal, as it moves the scope of the comment from the entire state variable into the target code.
- When running lff again on the above output, it removes the comment altogether:
  ```
  state grid: SnakeGrid = {= SnakeGrid::new(grid_side, &snake) =}
  ```

**Comment by petervdonovan** [comment]
As for point 1:

> In [HelloWorldPhysicalAfter.lf](https://github.com/lf-lang/examples-lingua-franca/blob/48f2e88887bc187600c06fd3a7d6cff9bd7f1d20/C/src/DistributedHelloWorld/HelloWorldPhysicalAfter.lf) the ; after at localhost was removed by lff. This, however, appears to be illegal syntax.

This issue has been discussed several times and I think the conclusion has been that we should remove the `at` grammar production entirely.

Points 2 and 3, and 6 will be fixed. It is possible that point 4 is related to point 3, but either way point 4 will be fixed.

Point 5 is controversial. I do not have a strong opinion about it, but I think it is worthwhile to make sure everyone agrees before implementing a fix. It is related to the first part of point 2, which I think is also controversial.

**Comment by cmnrd** [comment]
Yes, regarding the comments we should find a consensus. Personally, I would prefer a less intrusive formatting, i.e. let the programmer decide where to place comments and keep them there. But I am also Ok with leaving things as they are right now. As you point out correctly, the other issues are much more problematic.
